app/src/main/java/com/kms/cura/model/request/AppointmentModelResponse.java

```diff
@@ -0,0 +1,84 @@
+package com.kms.cura.model.request;
+
+import com.android.volley.VolleyError;
+import com.google.gson.Gson;
+import com.kms.cura.entity.AppointmentEntity;
+import com.kms.cura.entity.user.PatientUserEntity;
+import com.kms.cura.entity.user.UserEntity;
+import com.kms.cura.model.AppointmentModel;
+import com.kms.cura.utils.CurrentUserProfile;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by linhtnvo on 7/15/2016.
+ */
+public class AppointmentModelResponse implements EntityModelResponse {
+    boolean gotResponse;
+    boolean responseError = false;
+    String error;
+    List<AppointmentEntity> appts;
+
+    public List<AppointmentEntity> getAppts() {
+        return appts;
+    }
+
+    public void setAppts(List<AppointmentEntity> appts) {
+        this.appts = appts;
+    }
+
+    @Override
+    public void onErrorResponse(VolleyError error) {
+        responseError =  true;
+        this.error = error.getMessage();
+        gotResponse = true;
+    }
+
+    @Override
+    public void onResponse(String response) {
+        JSONObject jsonObject = null;
+        Gson gson = new Gson();
+        try {
+            jsonObject = new JSONObject(response);
+            boolean status = jsonObject.getBoolean(UserEntity.STATUS_KEY);
+            PatientUserEntity patient = (PatientUserEntity) CurrentUserProfile.getInstance().getEntity();
```

Comment:
Do not get patient here from CurrentUserProfile. Instead pass patient to the constructor and set to a field.